### ISSUE TITLE
On cart update tier prices with expired date are taken in total calculation

### ISSUE THREAD

**Issue by DegrizNet** [rukrlf/tier-price-date-range#2]
Hello,

first thank you for a great module. I tested it and found out when there is expired date and you go to chechkout/cart page on first load correct total is displayed but then when it reloads (customer/section/load/?sections=cart) with wrong totals (expired tier prices are taken into account).

[video-19275159-ca159b91aca993d98b464a029e16028b.webm](https://github.com/rukrlf/tier-price-date-range/assets/5030172/52b9e051-d1e3-403b-a830-229106e2dea0)

Magento ver. 2.4.5-p1 is in use and Slovenian language (I see that date format is notthe same on special price and tier price). Could that be the problem? Or can you point me in direction to where could be a problem? Thank you!

![Acerola-Izdelki-Inventory-Catalog-Magento-Admin](https://github.com/rukrlf/tier-price-date-range/assets/5030172/cad7dba1-3453-426f-9c8e-459dc3ae9bef)

**Comment by rukrlf** [comment]
Hello,

Glad you tried the module.
I'm not quite sure of the exact point to be looked at right now. Also, I'm
a bit busy with some work currently.
Will try to find the cause of the issue, but can't guarantee a specific
date. If you find a solution, pls create a PR.

Cheers!
Rukshan

**Comment by DegrizNet** [comment]
I understand, was first waiting on your replay to see if there is a quick solution. :) If I find a solution I will let you know.

**Comment by rukrlf** [comment]
Thank you!

**Comment by DegrizNet** [comment]
Sorry, but this was on hold for some time and just now I found some time to make changes. Probably needs some more testing to use on production but from my test it worked ok. Will update after testing if problems will be found.

Thank you again for your module.

**Comment by rukrlf** [comment]
Hi Anze,

Thanks for the work, much appreciated!

Cheers!
